pkg/mcproducer/scripts/pull-image.sh

```diff
@@ -1,15 +1,13 @@
 #!/bin/bash
 
 
-kernel_module_image_filepath="$KERNEL_MODULE_IMAGE_FILEPATH"
 worker_image="$WORKER_IMAGE"
 kernel_module_image="$KERNEL_MODULE_IMAGE"
 kernel_module_image_tag=$(uname -r)
 full_kernel_module_image="$kernel_module_image:$kernel_module_image_tag"
 
-if [ -e $kernel_module_image_filepath ]; then
-    echo "File $kernel_module_image_filepath found. Nothing to do, the file was handled, removing $kernel_module_image_filepath and $kmm_config_file_filepath"
-    rm -f $kernel_module_image_filepath
+if [ -n "$(podman images -q $worker_image 2> /dev/null)" ]; then
```

Comment:
i think you should check the kernel_module_image here, since ti it the second one pulled. in case you check the worker image, and the scripts fails on kernel_module image, then on the next reboot the kernel_module image will never be pulled